src/devices/Ds3231/Ds3231.cs

```diff
@@ -45,26 +45,18 @@ public Ds3231(I2cDevice sensor)
         /// <returns>DS3231 Time</returns>
         private DateTime ReadTime()
         {
+            // Sec, Min, Hour, Day, Date, Month & Century, Year
             Span<byte> rawData = stackalloc byte[7];
 
-            _sensor.Write(new [] { (byte)Register.RTC_SEC_REG_ADDR });
+            _sensor.WriteByte((byte)Register.RTC_SEC_REG_ADDR);
             _sensor.Read(rawData);
 
-            Ds3231Data data = new Ds3231Data();
-
-            data.Sec = Bcd2Int(rawData[0]);
-            data.Min = Bcd2Int(rawData[1]);
-            data.Hour = Bcd2Int(rawData[2]);
-            data.Day = Bcd2Int(rawData[3]);
-            data.Date = Bcd2Int(rawData[4]);
-            data.Month = Bcd2Int((byte)(rawData[5] & 0x1F));
-            data.Century = rawData[5] >> 7;
-            if (data.Century == 1)
-                data.Year = 2000 + Bcd2Int(rawData[6]);
-            else
-                data.Year = 1900 + Bcd2Int(rawData[6]);
-
-            return new DateTime(data.Year, data.Month, data.Date, data.Hour, data.Min, data.Sec);
+            return new DateTime(rawData[5] >> 7 == 1 ? 2000 + Bcd2Int(rawData[6]) : 1900 + Bcd2Int(rawData[6]),
```

Comment:
> `rawData[5] >> 7 == 1`

Aren't you missing a `& 1` just to check if the bit is turned on? Is there ever a case where rawData >> 7 can be more than 0 or 1 ?

Reply:
The right shift of byte type by 7 bits should not be greater than 1. Do you think this step should be added? If necessary, I'll add it.😄

Reply:
I think it's fine as is